src/bytes.rs

```diff
@@ -145,6 +145,39 @@ impl Bytes {
         }
     }
 
+    /// No cost BytesMut to Bytes converter
+    ///
+    /// # Safety
+    /// Make sure `bytes.kind()` is `KIND_VEC`
+    #[inline]
+    pub(crate) unsafe fn from_bytes_mut_vec(bytes: BytesMut, off: usize) -> Bytes {
+        if bytes.is_empty() {
+            return Bytes::new();
+        }
+
+        let off_ptr = bytes.as_ptr();
+        let ptr = off_ptr.offset(-(off as isize));
+        let len = bytes.len();
+        mem::forget(bytes);
+
+        if ptr as usize & KIND_MASK == 0 {
+            let data = ptr as usize | KIND_VEC;
+            Bytes {
+                ptr: off_ptr,
+                len,
+                data: AtomicPtr::new(data as *mut _),
+                vtable: &PROMOTABLE_EVEN_VTABLE,
+            }
+        } else {
+            Bytes {
+                ptr: off_ptr,
+                len,
+                data: AtomicPtr::new(ptr as *mut _),
+                vtable: &PROMOTABLE_ODD_VTABLE,
+            }
+        }
```

Comment:
style nit, take it or leave it: this might be a little more readable as
```suggestion
        let (data, vtable) = if ptr as usize & KIND_MASK == 0 {
            (ptr as usize | KIND_VEC as *mut _, &PROMOTABLE_EVEN_VTABLE)
        } else {
            (ptr, &PROMOTABLE_ODD_VTABLE)
        };
        
        Bytes {
            ptr: off_ptr,
            len,
            data: AtomicPtr::new(data),
            vtable,
        }
```